### ISSUE TITLE
Comments with html can cause lint failure (no-empty-source)

### ISSUE THREAD

**Issue by metalsheep** [stylelint/stylelint#2493]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where stylelint fails to lint files with html in a commented line of SCSS when using the standard configs.

> Which rule, if any, is this issue related to?

no-empty-source

> What CSS is needed to reproduce this issue?

```scss
a {
  display: inline;
}

// <div>
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "processors": ["stylelint-processor-html"],
  "extends": "stylelint-config-standard",
}
```

> Which version of stylelint are you using?

7.10.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI via:
```
▶ ./node_modules/stylelint/bin/stylelint.js ./test.scss
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes it appears to be related to SCSS comments

> What did you expect to happen?

The empty source rule would not trigger, and the file would be linted ignoring the comments

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
▶ ./node_modules/stylelint/bin/stylelint.js ./test.scss

test.scss
 1:1  ✖  Unexpected empty source   no-empty-source
```

**Comment by evilebottnawi** [comment]
@metalsheep please use issue template

**Comment by metalsheep** [comment]
@evilebottnawi updated to include the template

**Comment by evilebottnawi** [comment]
@metalsheep thanks, tomorrow i can investigate this, feel free to investigate this self and help our project:+1:

**Comment by evilebottnawi** [comment]
@metalsheep This behavior occurs because `scss` enter the processor, but it expects html. Disable `stylelint-processor-html` for `scss` files.

@stylelint/core what do you this it is excepted or not? What if i want to linting in html files and css files? Maybe we can passing filename to processor, it is will allow processor skip parse unexpected files?

**Comment by jeddy3** [comment]
> Maybe we can passing filename to processor? 

I believe [we already do that](https://github.com/mapbox/stylelint-processor-markdown/blob/master/index.js#L14).

> What if i want to linting in html files and css files? 

As such, I think this can either be resolved in `stylelint-processor-html` by skipping over anything other than `html` and `vue` files. That might be too restrictive, though.

Or we can update the the [processors documentation](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#processors) to recommend that separate stylelint task (with it's own config) is used when using a processor e.g.

```shell
stylelint "**/*.css"
stylelint "**./*.html --config stylelint-html.config.json
```

```js
// stylelint.config.json
{
  "extends": ["stylelint-config-standard"]
}
```

```js
// stylelint-html.config.json
{
  "extends": ["stylelint.config.js"],
  "processors": ["stylelint-processor-html"]
}
```

**Comment by davidtheclark** [comment]
Yep, I think this is an issue for `stylelint-processor-html`, not stylelint itself.
